app/models/katello/erratum.rb

```diff
@@ -46,10 +46,8 @@ def self.repository_association_class
     end
 
     def self.applicable_to_hosts(hosts)
-      self.joins(:content_facet_errata).joins(:content_facets).
-         where("#{Katello::Host::ContentFacet.table_name}.host_id" => hosts).
-         select("DISTINCT ON (#{self.table_name}.updated, #{self.table_name}.id) #{self.table_name}.*").
-         order("#{self.table_name}.updated desc, #{self.table_name}.id")
+      self.joins(:content_facets).
+        where("#{Katello::Host::ContentFacet.table_name}.host_id" => hosts).uniq
```

Comment:
One idea, but not sure how good it is:

Have a jenkins test that runs once a week and runs bats, but runs a new bats test that loads up a Katello with data and exercises it a bit, then checks for any egregiously slow operations.